### PR TITLE
Add /c/kn/uuid-uuid-uuid/receive style endpoints for all channel types

### PR DESCRIPTION
**Pull request by nicpottier**
We'll still have to do nginx remappings ourselves, but future peeps will be able to just reroute all of /c/ to courier via a application level load balancer.

### CODE REVIEW

**Comment by nicpottier** [comment]
@rowanseymour review?

**Comment by nicpottier** [comment]
@norkans7

**Comment by nicpottier** [comment]
This is needed in order for us to move to twilio as well as due to signatures our URLs need to match between courier and rapidpro

File: temba/channels/handlers.py [hunk 1]
```diff
@@ -323,8 +339,11 @@ def post(self, request, *args, **kwargs):
 
 class AfricasTalkingHandler(BaseChannelHandler):
 
-    url = r'^africastalking/(?P<action>delivery|callback)/(?P<uuid>[a-z0-9\-]+)/$'
-    url_name = 'handlers.africas_talking_handler'
+    courier_url = r'^at/(?P<uuid>[a-z0-9\-]+)/(?P<action>receive|delivery|callback|status)$'
```
Comment (rowanseymour) [inline temba/channels/handlers.py]: Is courier ready to handle old action names?

File: temba/channels/handlers.py [hunk 2]
```diff
@@ -1783,9 +1843,9 @@ def post(self, request, *args, **kwargs):
 
 
 class StartHandler(BaseChannelHandler):
-
-    url = r'^start/(?P<action>receive)/(?P<uuid>[a-z0-9\-]+)/?$'
-    url_name = 'handlers.start_handler'
+    courier_url = r'^st/(?P<uuid>[a-z0-9\-]+)/(?P<action>receive)$'
```
Comment (rowanseymour) [inline temba/channels/handlers.py]: no courier_name ?

File: temba/channels/urls.py [hunk 3]
```diff
@@ -11,11 +11,15 @@
 
 claim_page_urls = [ch_type.get_claim_url() for ch_type in Channel.get_types() if ch_type.claim_view]
 
+courier_urls = []
 handler_urls = []
+
 for handler in get_channel_handlers():
-    rel_url, url_name = handler.get_url()
-    handler_urls.append(url(rel_url, handler.as_view(), name=url_name))
+    rel_url, url_name = handler.get_courier_url()
```
Comment (rowanseymour) [inline temba/channels/urls.py]: Not all handlers have courier urls, no?

File: temba/channels/handlers.py (hunk 1 above)
Comment (nicpottier) [reply temba/channels/handlers.py]: For those that are different we can normalize them via nginx or yes, add support for them.

File: temba/channels/urls.py (hunk 3 above)
Comment (nicpottier) [reply temba/channels/urls.py]: Doh, good catch.

File: temba/channels/handlers.py (hunk 2 above)
Comment (nicpottier) [reply temba/channels/handlers.py]: Added.

**Pull request merged**